### PR TITLE
Book: Add internal section for crate-organization

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
We've discussed this many times. Let's finally put it in writing.

### CODE REVIEW

File: book/src/internals/crate-organization.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+Limitation: The `cxx-qt` crate itself is limited to implementing a superset of CXX.
+This means its goal is to add additional Qt-specific features to CXX.
+Namely support for Qts Meta-Object-System, which includes QObjects, QEnums, properties, signals, slots, etc.
+
+CXX-Qt aims to **allow you** to wrap Qt APIs and create your own Qt-compatible types.
+It **does not** aim to actually wrap Qt API.
+Apart from a few extremely basic types that are needed for this goal, wrapping the actual Qt API is delegated to `cxx-qt-lib` and `cxx-qt-lib-extras`.
+
+This should also allow `cxx-qt-lib` and `cxx-qt-lib-extras` to be used without the user needing to use cxx-qt itself, which is why they're explicitly seperated.
```
Comment (ahayzen-kdab) [inline book/src/internals/crate-organization.md:40]: There is cxx-qt vs CXX-Qt, this sentence is using lowercase version compared to others using upper case

Comment (LeonMatthesKDAB) [reply book/src/internals/crate-organization.md]: Right, I'm mostly using the lower-case code-block version when referring to the specific crate.

And CXX-Qt to refer to the whole framework/concept.
I tried updating the PR to make this more consistent.

File: book/src/internals/crate-organization.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+>
+> Specifically, we may simply drop a binding from the crate if it causes issues.
+
+> ⚠️  Bindings may be incomplete or undocumented!
+
+> ⚠️  Memory Safety Errors and other issues are possible!
+
+Basically: There be dragons 🐉 - use at your own risk.
+
+Forking of `cxqt-lib-extras` is encouraged to create something stable you can rely on for yourself.
```
Comment (ahayzen-kdab) [inline book/src/internals/crate-organization.md:135]: typo, should be `cxx-qt-lib-extras` ?

Comment (LeonMatthesKDAB) [reply book/src/internals/crate-organization.md]: Fixed

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**